### DISCUSSION TITLE
How about an initramfs-only variant?

### DISCUSSION THREAD

**Discussion by cipriancraciun** [finnix/finnix#51] [Ideas]
When I used to use SystemRescueCD, I always started it with `docache` (and then in later versions with `copytoram`) that copied the SquashFS image into RAM, so that I can then safely remove the USB stick and be sure I'll not damage it, and also to have a snappier system.  (To my knowledge, at least in variant 125 of Finnix there isn't such an option.)

Thus, I wanted to open this question to the Finnix community, about having a variant of Finnix that is composed only from a kernel (the current kernel) and an initramfs that holds the current `filesystem.squashfs` equivalent.  (I.e. there is no need for the SquashFS image anymore;  we don't need to recreate the SystemRescueCD `docache` feature, we just obtain something functionally equivalent.)

We wouldn't need to have disk images, just the two files for download.

This would be something similar to what many Linux distributions provide as a netboot.

----

What would be the benefit of such a feature?

* first of all, one can safely remove the media once the system has booted;
* also, as hinted above with the netboot, one could use iPXE (or similar) to launch Finnix from the network;
* my special use-case would be to have a UKI (Unified Kernel Image) that I can deploy in UEFI environments as a single EFI executable;

----

How would such a feature be implemented?

My initial take would be to just take the current `filesystem.squashfs` and transform it into a `cpio` and then use that as an initrd.

However, and here @rfinnie can provide useful feedback as he knows the packaging better than anyone, I think extending the current packaging scripts to also create the `filesystem.cpio` variant would be enough to solve it.

**Comment by sverrips** [comment]
What's wrong with clicking "e" at the boot menu (grub2) and adding an option (as text) "toram" to the initrd.img line?

**Reply by cipriancraciun** [reply]
> What's wrong with clicking "e" at the boot menu (grub2) and adding an option (as text) "toram" to the initrd.img line?

Although it solves the issue of removing the boot media after boot, (I didn't know about the `toram` option), it doesn't solve the other two use-cases I've mentioned:
* PXE or other network-boot-like mechanisms;  (where we can't emulate disks;)
* UKI (Unified Kernel Images) that simplify the UEFI Secure Boot process;

**Reply by cipriancraciun** [reply]
Yesterday I had the opportunity to test the `toram` option of the live CD, and although it might perhaps work, it doesn't quite work as other live distribution alternatives (especially SysRescueCD), namely:
* (expected behaviour) in case of SysRescueCD, one can specify where the SquashFS image lies on the boot media, and only that file is copied to ram;
* (observed behaviour) however in case of Finnix (via the Debian LiveCD implementation), it tries to copy into ram the **whole contents of the boot media**;

Why is that a problem:  for example in case one wants to build a multi-purpose rescue boot stick, which contains other things besides Finnix (like in my case multiple variants of SysRescueCD, OpenSUSE netboot, etc.), when booting Finnix with the `toram` option, it ends up copying many gigabytes into memory, which will eventually run out of.

Thus, why I've said an initramfs-only variant might solve this (and other issues).
